backend/src/tests/demo_db/conftest.py

```diff
@@ -0,0 +1,159 @@
+
+ASYNC_TEST_DB_URL = f"cockroachdb+asyncpg://{TEST_DB_USER}:{TEST_DB_PASS}@{TEST_DB_HOST}:{TEST_DB_PORT}/{TEST_DB_NAME}"
+async_engine = create_async_engine(ASYNC_TEST_DB_URL)
+
+
+@pytest.fixture
+async def engine():
+    async with async_engine.connect() as conn:
+        await conn.begin()
+        await conn.begin_nested()
+        yield conn
+        await conn.rollback()
+
+
+@pytest.fixture
+async def db_session(engine):
+    # Need to put this async context manager definition inside the fixture
+    # in order to have access to the engine
+    # If we passed db_session into functions as FastAPI dependency injection
+    # then this would look different (and probably cleaner)
+    @asynccontextmanager
+    async def test_db_session() -> AsyncIterator[AsyncSession]:
+        Session = sessionmaker(engine, class_=AsyncSession, expire_on_commit=False)
+        session = Session()
+        try:
+            yield session
+            await session.commit()
+        finally:
+            await session.close()
+
+    yield test_db_session
+
+
+# 3. Fixture for creating a seed test user
+# This is used in authed_client and also when other tests need
+# to create seed data attached to a user (e.g. Todo tests)
+@pytest.fixture
+async def fake_user(db_session: AsyncIterator[AsyncSession]):
+    "Fixture for creating a seed test user"
+    user_data = {"name": "test_user", "password": "pass"}
+    async with db_session() as session:
+        user = await UserDAO.create(session, user_data)
+    yield user
+
+
+# 4. httpx.AsyncClient and a second version that has the Authorization header
+# already set up to be logged in as the test user
+@pytest.fixture
+async def client():
+    "Fixture to return an HTTP client for the fastapi app"
+    async with httpx.AsyncClient(
+        app=app, base_url="http://test", follow_redirects=True
+    ) as client:
+        yield client
+
+
+@pytest.fixture
+async def authed_client(client: httpx.AsyncClient, fake_user: UserDAO):
+    """
+    Fixture to create an HTTP client that includes
+    the Authorization header to authenticate against
+    FastAPI endpoints, using the "test_user" credentials
+    """
+    auth_header = {"Authorization": f"bearer {fake_user.name}"}
+    client.headers.update(auth_header)
+    yield client
+
+
+# 5. Start in-memory cockroach database and create the tables
+# mirakuru is a library that can start processes and give you ways to tell when they are up
+SYNC_TEST_DB_URL = f"cockroachdb://{TEST_DB_USER}:{TEST_DB_PASS}@{TEST_DB_HOST}:{TEST_DB_PORT}/{TEST_DB_NAME}?sslmode=require"
+sync_engine = create_engine(SYNC_TEST_DB_URL)
+
+
+@pytest.fixture(scope="session")
+def manage_cockroach():
+    print("in manage_cockroach")
+    with mirakuru.TCPExecutor(
+        COCKROACH_CMD, host=TEST_DB_HOST, port=int(TEST_DB_PORT)
+    ) as process:
+        assert process.running()
+        print("Cockroach is running according to mirakuru.TCPExecutor")
```

Comment:
Suggest moving the (sleep .1; try to connect and fire off "select true") loop into here, manage_cockroach(), after the process.running() initial assertion so that this won't yield until is a fully scurrying bugger. Makes create_tables() then a bit more 'pure', and shouldn't need any loop or whatnot. Just bitblasts through creating the tables.

Reply:
I agree with the sentiment.  I think leaving this here is okay now with a plan to PR back to mirakuru with an executor that better fits our needs.  Is there an easy sqlalchemy command to just test login?  Like `sa.text("SELECT 1")` or something?